metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_source.py

```diff
@@ -0,0 +1,607 @@
+                )
+                for ds in query.queried_datasets
+            ]
+
+            downstream_urn = make_dataset_urn_with_platform_instance(
+                platform=make_data_platform_urn(self.get_platform()),
+                name=f"dremio.{query.affected_dataset.lower()}",
+                env=self.config.env,
+                platform_instance=self.config.platform_instance,
+            )
+
+            # Add query to SqlParsingAggregator
+            self.sql_parsing_aggregator.add_known_query_lineage(
+                KnownQueryLineageInfo(
+                    query_text=query.query,
+                    upstreams=upstream_urns,
+                    downstream=downstream_urn,
+                ),
+                merge_lineage=True,
+            )
+
+        # Add observed query
+        self.sql_parsing_aggregator.add_observed_query(
+            ObservedQuery(
+                query=query.query,
+                timestamp=query.submitted_ts,
+                user=CorpUserUrn(username=query.username),
+                default_db=self.default_db,
+            )
+        )
+
+    def _map_dremio_dataset_to_urn(
+        self,
+        dremio_source: str,
+        dremio_path: List[str],
+        dremio_dataset: str,
+    ) -> Optional[str]:
+        """
+        Map a Dremio dataset to a DataHub URN.
+        """
+        mapping = self.source_map.get(dremio_source.lower())
+        if not mapping:
+            return None
+
+        platform = mapping.get("platform")
+        if not platform:
+            return None
+
+        platform_instance = mapping.get(
+            "platform_instance", self.config.platform_instance
+        )
+        env = mapping.get("env", self.config.env)
+
+        root_path = ""
+        database_name = ""
+
+        if mapping.get("dremio_source_type") == "file_object_storage":
+            if mapping.get("root_path"):
+                root_path = f"{mapping['root_path'][1:]}/"
+            dremio_dataset = f"{root_path}{'/'.join(dremio_path[1:])}/{dremio_dataset}"
+        else:
+            if mapping.get("database_name"):
+                database_name = f"{mapping['database_name']}."
+            dremio_dataset = (
+                f"{database_name}{'.'.join(dremio_path[1:])}.{dremio_dataset}"
+            )
+
+        if platform_instance:
+            return make_dataset_urn_with_platform_instance(
+                platform=platform.lower(),
+                name=dremio_dataset,
+                platform_instance=platform_instance,
+                env=env,
+            )
+
+        return make_dataset_urn_with_platform_instance(
+            platform=platform.lower(),
+            name=dremio_dataset,
+            platform_instance=None,
+            env=env,
+        )
+
+    def get_report(self) -> SourceReport:
+        """
+        Get the source report.
+        """
+        return self.report
+
+    def close(self) -> None:

```

Comment:
Either remove this method or add `super().close()` to it to avoid issues with stateful ingestion base close.

Reply:
Removed